dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/cdi/RepositoryProducer.java

```diff
@@ -75,32 +75,93 @@ public class RepositoryProducer<R> implements Producer<R>, ProducerFactory<R>, B
      */
     private static final long INIT_TIMEOUT_SEC = TimeUnit.MINUTES.toSeconds(5);
 
+    /**
+     * Qualifiers for repository bean instances.
+     */
     private static final Set<Annotation> QUALIFIERS = //
                     Set.of(Any.Literal.INSTANCE, Default.Literal.INSTANCE);
 
+    /**
+     * CDI bean manager.
+     */
     private final BeanManager beanMgr;
+
+    /**
+     * A set of the repository interface type.
+     */
     private final Set<Type> beanTypes;
-    private final FutureEMBuilder futureEMBuilder;
+
+    /**
+     * Future for the EntityManagerBuilder.
+     */
+    private FutureEMBuilder futureEMBuilder;
+
+    /**
+     * CDI extension for the Jakarta Data provider.
+     */
     private final DataExtension extension;
+
+    /**
+     * Map of intercepted instance to bean instance. Applies when a repository
+     * interface has methods with interceptor annotations, such as
+     * Asynchronous or Transactional.
+     */
     private final Map<R, R> intercepted = new ConcurrentHashMap<>();
-    private final Class<?> primaryEntityClass;
+
+    /**
+     * Primary entity class, if any, for the repository.
+     */
+    private Class<?> primaryEntityClass;
+
+    /**
+     * Jakarta Data provider.
+     */
     private final DataProvider provider;
-    private final Map<Class<?>, List<QueryInfo>> queriesPerEntityClass;
+
+    /**
+     * Information about each repository method, grouped by entity class.
+     */
+    final Map<Class<?>, List<QueryInfo>> queriesPerEntityClass;
+
+    /**
+     * Handler for the most recently created repository bean instance.
+     */
     private final AtomicReference<RepositoryImpl<?>> repositoryImplRef = //
-                    new AtomicReference<>(); // most recently created instance
+                    new AtomicReference<>();
+
+    /**
+     * The repository interface.
+     */
     private final Class<?> repositoryInterface;
 
-    RepositoryProducer(Class<?> repositoryInterface, BeanManager beanMgr, DataProvider provider, DataExtension extension,
-                       FutureEMBuilder futureEMBuilder, Class<?> primaryEntityClass, Map<Class<?>, List<QueryInfo>> queriesPerEntityClass) {
+    /**
+     * Indicates if the repository is stateful (true) or stateless (false).
+     */
+    private boolean stateful;
```

Comment:
```suggestion
     * Indicates if the repository is stateful (true) or stateless (false).
     * Default: false
     */
    private boolean stateful = false;
```

nitpick: since this isn't set to a default value in the constructor I would find it more readable to set the default value here (even though booleans default to false it can be easy to forget that :/ )